avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -691,9 +691,6 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
       case STRING:
         if (enhancedSchemaSupport && schema.parameters() != null
             && schema.parameters().containsKey(AVRO_TYPE_ENUM)) {
-          String enumSchemaName = schema.parameters().get(AVRO_TYPE_ENUM);
-          org.apache.avro.Schema cachedEnum = schemaMap.get(enumSchemaName);
-          if (cachedEnum == null) {
```

Comment:
Maybe I'm misunderstanding it, but I believe my presumption was that the top-level Connect schema would be cached, not the internal enum one.  On first pass, the top-level schema would be cached (line 853). On subsequent passes, we would get cache hits on schemaMap, entailing the cached internal enum schema.